Switch msedge version detection to use CFBundleShortVersionString

vanderboon opened the pull request:
For msedge the current version is pulled from `CFBundleVersion`, this value does not match the version of Edge browser as reported by the browser itself.  `CFBundleShortVersionString` does have the same value as reported by the browser.

e.g.
![image](https://github.com/user-attachments/assets/6bbbd518-e581-4fb9-8a68-54258cabe61c)

With `CFBundleVersion`:
```
print(json.dumps(list(browsers.browsers()), indent=4)
... )
[
    {
        "browser_type": "msedge",
        "path": "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge",
        "display_name": "Microsoft Edge",
        "version": "133.3065.25022682"
    }
]
```

With `CFBundleShortVersionString`:
```

print(json.dumps(list(browsers.browsers()), indent=4))
[
    {
        "browser_type": "msedge",
        "path": "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge",
        "display_name": "Microsoft Edge",
        "version": "133.0.3065.82"
    }
]
```

From the [Apple developer docs `CFBundleShortVersionString`](https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleshortversionstring): _This key is used throughout the system to identify the version of the bundle._

And from the [Apple developer docs `CFBundleVersion `](https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleversion): _This key is required by the App Store and is used throughout the system to identify the version of the build. For macOS apps, increment the build version before you distribute a build._

It seems more appropriate to use `CFBundleShortVersionString`.

vanderboon committed:
Switch msedge version detection to use CFBundleShortVersionString

Signed-off-by: vanderboon <martin@vanderboon.co.uk>

roniemartinez commented:
Thanks @vanderboon 

`CFBundleShortVersionString` actually makes more sense for all of them (even `KSVersion` is the same as `CFBundleShortVersionString`). Do you mind removing the version strings altogether and hard-code `version = plist["CFBundleShortVersionString"]`?

vanderboon commented:
I checked Opera & Firefox and the output matches the About _Browser_ version.

I did check the Brave browser and it doesn't report any matching version.

About Brave:
<img width="490" alt="image" src="https://github.com/user-attachments/assets/74e06e7b-fac5-4362-85ae-0b51bf67ba5a" />

`CFBundleVersion`
<img width="209" alt="image" src="https://github.com/user-attachments/assets/09a6d78e-3153-4287-9a33-a0c9fd2bcb71" />

`CFBundleShortVersionString`
<img width="289" alt="image" src="https://github.com/user-attachments/assets/24e5c39a-4ee0-47bb-b463-800106a8ba04" />

`KSVersion`
<img width="252" alt="image" src="https://github.com/user-attachments/assets/c3703845-a114-441d-95f1-485ad2c4f92e" />

Current `browsers()`:
```
print(json.dumps(list(browsers.browsers()), indent=4))
[
    {
        "browser_type": "brave",
        "path": "/Applications/Brave Browser.app/Contents/MacOS/Brave Browser",
        "display_name": "Brave Browser",
        "version": "175.181"
    }
]
```

None of the above match 🤷 
Hard-coding `version = plist["CFBundleShortVersionString"]` would change the Brave output to something that doesn't match the About Brave output.  The closest is `CFBundleVersion` but with some splitting to force the output to `1.75.181`. 

I suggest:
- switch Chrome & Edge to `CFBundleShortVersionString`
- leave Firefox using `CFBundleShortVersionString`
- leave Opera using `CFBundleVersion`
- update Brave to use a split `CFBundleVersion`

What's your preference with this?  Happy to update accordingly.

roniemartinez commented:
>  switch Chrome & Edge to CFBundleShortVersionString
    leave Firefox using CFBundleShortVersionString
    leave Opera using CFBundleVersion

Thanks for checking. Yes, let's do these.

> update Brave to use a split CFBundleVersion

Not sure actually, let's keep the existing. I think we might mess up the split. (Unless the format is documented somewhere)

vanderboon commented:
> Not sure actually, let's keep the existing. I think we might mess up the split. (Unless the format is documented somewhere)

Looks like it is documented...
https://github.com/brave/brave-core/blob/cd1c1fdf88965b409bea4a1cd87aa4638e21355b/build/mac/tweak_info_plist.py#L46-L58

They take the version from https://github.com/brave/brave-core/blob/cd1c1fdf88965b409bea4a1cd87aa4638e21355b/package.json#L3 and use `(major+100).minor.build`.

That can be reversed.  If you're happy with that I'll update the PR.

roniemartinez commented:
Awesome! Those are helpful. Yes please update the PR. Thank you

vanderboon committed:
Reverse version manipulation done by the Brave browser on osx

vanderboon commented:
PR updated...

- Switched Chrome to use `CFBundleShortVersionString`
- Switched Edge to use `CFBundleShortVersionString`
- Added a function to reverse the version manipulation done by the Brave browser - https://github.com/brave/brave-core/blob/master/build/mac/tweak_info_plist.py#L46-L58

Chrome:
<img width="372" alt="image" src="https://github.com/user-attachments/assets/6e968520-d2f4-4bee-8e6a-9ee2f5703bfb" />

Chromium:
<img width="464" alt="image" src="https://github.com/user-attachments/assets/ea1169f1-462a-4b72-a4f2-1ea5bae8ca81" />

Firefox:
<img width="317" alt="image" src="https://github.com/user-attachments/assets/51d68def-bb06-46cb-ba29-e8bddd594554" />

Safari:
<img width="185" alt="image" src="https://github.com/user-attachments/assets/32a464b1-97d3-4f0d-ac84-4e7a3b0aadbf" />

Edge:
<img width="321" alt="image" src="https://github.com/user-attachments/assets/9178a91e-1d6c-4525-886a-600b64a5aa65" />

Brave:
<img width="518" alt="image" src="https://github.com/user-attachments/assets/758f75a2-05bc-402e-ab53-44a47fbb3724" />

```
print(json.dumps(list(browsers.browsers()), indent=4))
[
    {
        "browser_type": "chrome",
        "path": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
        "display_name": "Google Chrome",
        "version": "133.0.6943.142"
    },
    {
        "browser_type": "chromium",
        "path": "/Applications/Chromium.app/Contents/MacOS/Chromium",
        "display_name": "Chromium",
        "version": "136.0.7050.0"
    },
    {
        "browser_type": "firefox",
        "path": "/Applications/Firefox.app/Contents/MacOS/firefox",
        "display_name": "Firefox",
        "version": "136.0"
    },
    {
        "browser_type": "safari",
        "path": "/Applications/Safari.app",
        "display_name": "Safari",
        "version": "18.3"
    },
    {
        "browser_type": "msedge",
        "path": "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge",
        "display_name": "Microsoft Edge",
        "version": "133.0.3065.92"
    },
    {
        "browser_type": "brave",
        "path": "/Applications/Brave Browser.app/Contents/MacOS/Brave Browser",
        "display_name": "Brave Browser",
        "version": "1.75.181"
    }
]
```

vanderboon committed:
Formatted

roniemartinez commented:
@vanderboon Thanks for the update. 🚀

roniemartinez commented:
@all-contributors please add @vanderboon for code
